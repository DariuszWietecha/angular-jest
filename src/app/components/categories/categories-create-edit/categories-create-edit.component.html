<h5><span class="text-capitalize">{{ type }}</span> category</h5>

<form [formGroup]="createEditForm" (ngSubmit)="onSubmit()">
  <div>
    <dt>
    <label for="name">
      Name:
    </label>
  </dt>
  <dd>
    <input id="name" type="text" formControlName="name">
  </dd>
  </div>

  <button class="button" type="submit"  class="btn btn-primary">Save</button>
</form>