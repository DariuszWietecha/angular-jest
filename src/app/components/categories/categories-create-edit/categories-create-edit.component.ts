import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { FormBuilder, FormGroup } from '@angular/forms';
import { CategoriesService } from '../../../services/categories.service';
import { Router } from '@angular/router';

@Component({
  selector: 'app-categories-create-edit',
  templateUrl: './categories-create-edit.component.html',
  styleUrls: ['./categories-create-edit.component.scss']
})
export class CategoriesCreateEditComponent implements OnInit {
  createEditForm: FormGroup;
  type: string;

  constructor(
    private route: ActivatedRoute,
    private formBuilder: FormBuilder,
    private router: Router,
    private categoriesService: CategoriesService
  ) {
    this.createEditForm = this.formBuilder.group({
      id: '',
      name: '',
    });
  }

  ngOnInit() {
    this.route.paramMap.subscribe(params => {
      this.type = params.get('type');

      if (this.type === 'edit') {
        this.categoriesService.get(params.get('id'))
          .subscribe((data) => {
            this.createEditForm.patchValue(data);
          });
      }
    });
  }

  onSubmit() {
    const method = this.type === 'edit' ? 'edit' : 'create';
    const formData = this.createEditForm.value;
    if (!formData.id) {
      delete formData.id;
    }

    this.categoriesService[method](formData)
      .subscribe(() => {
        this.router.navigate(['']);
      });

    this.createEditForm.reset();
  }
}
