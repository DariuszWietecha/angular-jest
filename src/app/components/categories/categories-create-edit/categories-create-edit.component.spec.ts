import { CategoriesCreateEditComponent } from './categories-create-edit.component';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { CategoriesService } from '../../../services/categories.service';
import { CompaniesService } from '../../../services/companies.service';
import { HttpClientModule } from '@angular/common/http';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';

describe('CategoriesCreateEditComponent', () => {
  let component: CategoriesCreateEditComponent;
  let fixture: ComponentFixture<CategoriesCreateEditComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientModule, ReactiveFormsModule, RouterTestingModule],
      declarations: [ CategoriesCreateEditComponent ],
      providers: [CategoriesService, CompaniesService],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CategoriesCreateEditComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapShot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
