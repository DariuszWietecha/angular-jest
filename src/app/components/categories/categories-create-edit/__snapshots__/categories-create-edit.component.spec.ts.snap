// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesCreateEditComponent snapShot 1`] = `
<app-categories-create-edit
  categoriesService={[Function CategoriesService]}
  createEditForm={[Function FormGroup]}
  formBuilder={[Function FormBuilder]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  type="null"
>
  <h4>
    <span
      class="text-capitalize"
    >
      
    </span>
     category
  </h4><form
    class="ng-untouched ng-pristine ng-valid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <div>
      <dt>
        <label
          for="name"
        >
           Name: 
        </label>
      </dt>
      <dd>
        <input
          class="ng-untouched ng-pristine ng-valid"
          formcontrolname="name"
          id="name"
          ng-reflect-name="name"
          type="text"
        />
      </dd>
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Save
    </button>
  </form>
</app-categories-create-edit>
`;
